feature/interface/singleton/ate_tests/singleton_test/singleton_test.go

```diff
@@ -198,6 +220,22 @@ func (tc *testCase) verifyInterfaceDUT(
 		t.Errorf("%s oper-status got %v, want %v", dp, got, opUp)
 	}
 
+	if speed, ok := portSpeed[dp.Speed()]; ok {
+		if tc.auto == forcedNegotiation || tc.auto == autoNegotiationWithDuplexSpeed {
+			if got := dip.Ethernet().PortSpeed().Get(t); got != speed {
+				t.Errorf("%s port-speed got %v, want %v", dp, got, speed)
+			}
+		}
+		if tc.auto == autoNegotiation || tc.auto == autoNegotiationWithDuplexSpeed {
+			if dip.Ethernet().AutoNegotiate().Get(t) {
```

Comment:
if `dip.Ethernet().AutoNegotiate().Get(t) == false`, that means the port does not support auto negotiation and auto negotiation is ignored, right? I guess in this case we still want to check port speed?

Reply:
The port-speed telemetry reflects only what the config sets, and it's checked above on line 225. On the other hand, the negotiated-port-speed is only set if the port supports auto negotiation.

In the autoNegotiationWithDuplexSpeed case, it will trigger both port-speed and negotiated-port-speed checks, so it's working as intended.